fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContextTest.java

```diff
@@ -0,0 +1,280 @@
+            assertEquals(readVersion1, readVersion2);
+            assertEquals(1, timer.getCount(FDBStoreTimer.Events.GET_READ_VERSION));
+            assertEquals(grvNanos, timer.getTimeNanos(FDBStoreTimer.Events.GET_READ_VERSION));
+        }
+    }
+
+    @Test
+    public void manyParallelReadVersions() throws InterruptedException, ExecutionException {
+        final FDBStoreTimer timer = new FDBStoreTimer();
+        try (FDBRecordContext context = fdb.openContext(null, timer)) {
+            List<CompletableFuture<Long>> futures = Stream.generate(context::getReadVersionAsync)
+                    .limit(10)
+                    .collect(Collectors.toList());
+            List<Long> readVersions = AsyncUtil.getAll(futures).get();
+            long firstReadVersion = readVersions.get(0);
+            for (long readVersion : readVersions) {
+                assertEquals(firstReadVersion, readVersion);
+            }
+            assertEquals(1, timer.getCount(FDBStoreTimer.Events.GET_READ_VERSION));
+        }
+    }
+
+    @Test
+    public void getReadVersionTimingWithInjectedLatency() {
+        final FDBStoreTimer timer = new FDBStoreTimer();
+        FDBDatabaseFactory factory = fdb.getFactory();
+        Function<FDBLatencySource, Long> oldLatencyInjector = factory.getLatencyInjector();
+        factory.setLatencyInjector(latencySource -> {
+            if (latencySource.equals(FDBLatencySource.GET_READ_VERSION)) {
+                return 50L;
+            } else {
+                return 10L;
+            }
+        });
+        factory.clear();
+        fdb = factory.getDatabase(fdb.getClusterFile());
+        try {
+            try (FDBRecordContext context = fdb.openContext(null, timer)) {
+                context.getReadVersion();
+            }
+            assertEquals(1, timer.getCount(FDBStoreTimer.Waits.WAIT_GET_READ_VERSION));
+            assertEquals(1, timer.getCount(FDBStoreTimer.Events.GET_READ_VERSION));
+            assertEquals(1, timer.getCount(FDBStoreTimer.Events.INJECTED_GET_READ_VERSION_LATENCY));
+
+            long waitNanos = timer.getTimeNanos(FDBStoreTimer.Waits.WAIT_GET_READ_VERSION);
+            long grvNanos = timer.getTimeNanos(FDBStoreTimer.Events.GET_READ_VERSION);
+            long injectedNanos = timer.getTimeNanos(FDBStoreTimer.Events.INJECTED_GET_READ_VERSION_LATENCY);
+
+            // Validate that the grvNanos includes the injected nanos. It is very likely that GRV is higher than
+            // the injected nanos, but theoretically could be different
+            assertThat(injectedNanos, lessThanOrEqualTo(grvNanos));
+            assertThat(waitNanos, greaterThan(0L));
+
+        } finally {
+            factory.setLatencyInjector(oldLatencyInjector);
+            factory.clear();
+        }
+    }
+
+    @ParameterizedTest(name = "closeWithOutstandingGetReadVersion [inject latency = {0}]")
+    @EnumSource(TestHelpers.BooleanEnum.class)
```

Comment:
`TestHelpers.BooleanEnum` already exists, but it seems much cleaner to have `@BooleanSource` that provides `true` and `false`

Reply:
Sure. Because this isn't included within JUnit by default, so I added one and switched the tests to use it instead of the boolean enum. That's isolated in its own commit: 7d58a4dc9ce0cf5.